src/main/java/com/cefriel/template/io/sql/SQLReader.java

```diff
@@ -94,27 +94,49 @@ public ResultSet executeQuery(String query) {
      * @return Result of the SQL query
      */
     public List<Map<String, String>> getDataframe(String query) {
-
         List<Map<String, String>> dataframe = new ArrayList<>();
+        String queryCheck = query.toLowerCase();
 
-        try (ResultSet resultSet = executeQuery(query)) {
-            int columnCount = resultSet.getMetaData().getColumnCount();
+        if(queryCheck.contains("select")) {
+            try (ResultSet resultSet = executeQuery(query)) {
+                int columnCount = resultSet.getMetaData().getColumnCount();
 
-            while (resultSet.next()) {
-                Map<String, String> row = new HashMap<>();
-                for (int i = 1; i <= columnCount; i++) {
-                    String columnName = resultSet.getMetaData().getColumnName(i);
-                    String columnValue = resultSet.getString(i);
-                    row.put(columnName, columnValue);
+                while (resultSet.next()) {
+                    Map<String, String> row = new HashMap<>();
+                    for (int i = 1; i <= columnCount; i++) {
+                        String columnName = resultSet.getMetaData().getColumnName(i);
+                        String columnValue = resultSet.getString(i);
+                        row.put(columnName, columnValue);
+                    }
+                    dataframe.add(row);
                 }
-                dataframe.add(row);
+            } catch (SQLException e) {
+                log.error(e.getMessage(), e);
             }
-        } catch (SQLException e) {
-            log.error(e.getMessage(), e);
         }
 
-        return dataframe;
+        else {
+            String q = "SELECT * FROM " + query;
```

Comment:
PreparedStatement should be initialised with `?` for the table name, then `setString ` should be invoked to add the table name

Reply:
This apparently cannot be done for table names. 
https://stackoverflow.com/questions/11312155/how-to-use-a-tablename-variable-for-a-java-prepared-statement-insert

Reply:
A reasonable thing to do might be to:
1. Retrieve all table names from the db connection
2. Check that the user submitted table name matches a table from the db
3. Signal an error if not

Reply:
That's a good idea! Maybe this can be done in the SQLReader constructor storing the list of tables as a private variable. In any case, we can create a separate issue for this problem mentioning also other security concerns due to the fact that a custom SQL query can be provided